### DISCUSSION TITLE
Ability to customise thumbnail sizes in the admin (ie entries display)

### DISCUSSION THREAD

**Discussion by MattAppleton** [craftcms/cms#11521] [Ideas]
We have starting working withCraft recently, and so far we love it — except …
The thumbnails our users see in entries are WAAAY too small!

![Screenshot 2022-04-19 at 16 51 25](https://user-images.githubusercontent.com/6295710/176401831-8109ce14-7870-4945-96dd-a1a358ebf2c3.png)

An example above for a site we are working on right now for some video editors. As you can see we just see a sequence of barely distinguishable brown rectangles. We will be migrating over a video portfolio site form Wordpress, where the main back-end function will be adding video reels, and the reordering them with drag and drop to create showcase video pages.

As you can imagine, being able to SEE a video still per entry is by far the quickest / easiest way to intuitively edit the showreels.

But this is impossible with Craft, the thumbnails are painfully small :(

What can we do? We need thumbnails _at least_ 2 times taller than they currently appear in the entries files (ideally more)...

Wordpress is painful in many ways, but the ability to customise the interface and in particular the thumbnail display in the backend is super useful and worth having!

**Comment by wsydney76** [comment]
Agree!

However you can add any content you want to the element index. (Found this in one of our recent Craft4 projects)

Add something like this to the `init` method of a (bootstrapped) custom module:

```php
use craft\base\Element;
use craft\elements\Entry;
use craft\events\ElementIndexTableAttributeEvent;
use craft\events\RegisterElementTableAttributesEvent;
use craft\events\SetElementTableAttributeHtmlEvent;
use craft\helpers\Html;

...

// Register element index column
Event::on(
    Entry::class,
    Element::EVENT_REGISTER_TABLE_ATTRIBUTES, function(RegisterElementTableAttributesEvent $event) {
    $event->tableAttributes['bigFeaturedImage'] = ['label' => Craft::t('site', 'Big Featured Image')];
});

// Column HTML
Event::on(
    Entry::class,
    Element::EVENT_SET_TABLE_ATTRIBUTE_HTML, function(SetElementTableAttributeHtmlEvent $event) {

    if ($event->attribute == 'bigFeaturedImage') {
        /** @var Entry $entry */
        $entry = $event->sender;
        $event->html = 'n/a';
        $event->handled = true;

        // Don't wait for transformed images if generateTransformsBeforePageLoad = true
        $config = Craft::$app->config->general;
        $oldSetting = $config->generateTransformsBeforePageLoad;
        $config->generateTransformsBeforePageLoad = false;

        if ($entry->featuredImage) {
            $image = $entry->featuredImage->one();
            if ($image) {
                $image->setTransform(['width' => 250, 'height' => 140]);
                $event->html = Html::tag('img', '', [
                    'src' => $image->url,
                    'style' => 'border-radius: 3px',
                    'width' => $image->width,
                    'height' => $image->height,
                    'alt' => $image->alt ?? $image->title,
                    'ondblclick' => "Craft.createElementEditor('craft\\\\elements\\\\Asset', {elementId: {$image->id}, siteId: {$entry->site->id}})"
                ]);
            }
        }

        $config->generateTransformsBeforePageLoad = $oldSetting;
    }
});

// Enable eager loading, requires Craft 4.1 so that fetching the assets with `->one()` will work.
Event::on(
	Entry::class,
	Entry::EVENT_PREP_QUERY_FOR_TABLE_ATTRIBUTE,
	function(ElementIndexTableAttributeEvent $event) {
		$query = $event->query;
		$attr = $event->attribute;

		if ($attr === 'bigFeaturedImage') {
			$query->andWith(
				['featuredImage', ['withTransforms' => [['width' => 250, 'height' => 140]]]]
			);
		}
	});

```

Instead of using EVENT_SET_TABLE_ATTRIBUTE_HTML, it looks cleaner to move that code to an entry behavior.

(Could be a bit slow though, didn't dig deep enough to find out whether this can be somehow eager loaded. And it does not support double clicking the image to open the slide out editor. But hey, it works.)  **

** EDIT
Added eager loading (`EVENT_PREP_QUERY_FOR_TABLE_ATTRIBUTE` event, from #9862 ) and opening the slide out editor (`ondblclick`)

/Mel

**Reply by MattAppleton** [reply]
@wsydney76 —  Very interesting, we haven't got into custom modules yet, but will be having a look for sure, thanks!!

**Reply by josh-attwood** [reply]
@wsydney76 Am I right that with the method you have above you would have to have a normal image size in the columns alongside this new, bigger version. If so do you have any examples of how we would retrieve the image from the entry rather than the column?

**Reply by wsydney76** [reply]
@josh-attwood In this example we have a custom asset field `featuredImage`, which has it's default column in the element index, that remains untouched.

And we add a custom column named `bigFeaturedImage` in the `EVENT_REGISTER_TABLE_ATTRIBUTES` event, and define the content it will have in the `EVENT_SET_TABLE_ATTRIBUTE_HTML`, fetching the image from our `featuredImage` field and resizing it.

So when you customize your element index columns, you will see both the default and the custom column and select only your custom one:

![image](https://user-images.githubusercontent.com/60010597/178228894-16e95d51-8ccd-4a74-93c7-49880477c6b8.png)
